packages/components/src/lib/hooks.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  bindings,
+  environment,
+  jsImports,
+  run,
+  runPartial,
+} from "@quri/squiggle-lang";
+import { useEffect, useMemo, useRef } from "react";
+
+type SquiggleArgs<T extends ReturnType<typeof run | typeof runPartial>> = {
+  code: string;
+  bindings?: bindings;
+  jsImports?: jsImports;
+  environment?: environment;
+  onChange?: (expr: Extract<T, { tag: "Ok" }>["value"] | undefined) => void;
+};
+
+const useSquiggleAny = <T extends ReturnType<typeof run | typeof runPartial>>(
+  args: SquiggleArgs<T>,
+  f: (...args: Parameters<typeof run>) => T
```

Comment:
Would appreciate it if you renamed `f` to `squiggleRunFunction` or something similar.

I'd love there to be only one run function IMO, but that would require some larger changes to how the interpreter works to get there.

Reply:
Yep, I tried to call `run` instead of `runPartial` but turned out that `run` converts its response with `createTsExport` so it doesn't work for bindings. Maybe there could be an option for `run` as to whether to convert the result? I think that'd be enough.